GitFyle.Core.Api/Services/Foundations/Configurations/ConfigurationService.Validations.cs

```diff
@@ -0,0 +1,142 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using GitFyle.Core.Api.Models.Foundations.Configurations;
+using GitFyle.Core.Api.Models.Foundations.Configurations.Exceptions;
+
+namespace GitFyle.Core.Api.Services.Foundations.Configurations
+{
+    internal partial class ConfigurationService
+    {
+        private async ValueTask ValidateConfigurationOnAdd(Configuration configuration)
+        {
+            ValidateConfigurationIsNotNull(configuration);
+
+            Validate(
+                (Rule: await IsInvalidAsync(configuration.Id), Parameter: nameof(configuration.Id)),
+                (Rule: await IsInvalidAsync(configuration.Name), Parameter: nameof(configuration.Name)),
+                (Rule: await IsInvalidAsync(configuration.Type), Parameter: nameof(configuration.Type)),
+                (Rule: await IsInvalidAsync(configuration.Value), Parameter: nameof(configuration.Value)),
+                (Rule: await IsInvalidAsync(configuration.CreatedBy), Parameter: nameof(configuration.CreatedBy)),
+                (Rule: await IsInvalidAsync(configuration.UpdatedBy), Parameter: nameof(configuration.UpdatedBy)),
+                (Rule: await IsInvalidAsync(configuration.CreatedDate), Parameter: nameof(configuration.CreatedDate)),
+                (Rule: await IsInvalidAsync(configuration.UpdatedDate), Parameter: nameof(configuration.UpdatedDate)),
+
+                (Rule: await IsValuesNotSameAsync(
```

Comment:
Update with renaming

```suggestion
                (Rule: await IsNotSameAsync(
```

Reply:
Done.